### ISSUE TITLE
Unusable If Launched When Internet Down

### ISSUE THREAD

**Issue by hammypants** [rust-analyzer/rust-analyzer#9613]
If a local network is available but the internet is down, RA w/ VSCode fails during bootstrapping, providing an error message that an outgoing request to the GitHub API has failed, rendering the extension unusable.

**Comment by bjorn3** [comment]
This only happens when vscode updated the extension before internet becomes unavailable. When launching the new version of the extension, it notices that the rust-analyzer binary doesn't match it's own version and requires the new version to be downloaded.

**Comment by lnicola** [comment]
@hammypants can you confirm that this only happens after an update?

**Comment by hammypants** [comment]
> @hammypants can you confirm that this only happens after an update?

Seems that way from my contrived testing.

**Comment by lnicola** [comment]
Then it seems like an edge case: you were able to download and install the extension update, but not the new server binary.

**Comment by hammypants** [comment]
I suppose. That's not really for me to decide. :D The circumstance [for me] of "the extension has automatically updated, but the IDE has not been relaunched" is quite likely, considering the uptime of the IDE is often many days. Unfortunately, the additional circumstance of "the internet has disappeared" is not _that_ uncommon, considering my ISP is Comcast. :haroldcringeemojihere:

I suppose the recommended change on my end here will be to disable automatic updates of the extension?

**Comment by lnicola** [comment]
Oh, does Code not reload extensions after an update? I was actually wondering how that worked...

> I suppose the recommended change on my end here will be to disable automatic updates of the extension?

Are you really running into this so often?

**Comment by bjorn3** [comment]
> Oh, does Code not reload extensions after an update?

You have to manually click on reload in the extension pane. This needs to reload the whole vscode window, so it better not happen automatically.

**Comment by lnicola** [comment]
Yeah, I thought it maybe updates them on start-up, before initialization.

**Comment by hammypants** [comment]
> Are you really running into this so often? 😕

I mean... 😞 

I'm assuming there's no way to tie "get new RA binary" to the VSC extension update, unless Microsoft is providing a hook there. Which would mean this is likely unsolvable. And even then, it may be architecturally painful. I don't expect RA to pretzel to fix this. I just thought I would log it in case something could be done, or if there could be a way to do something in the future.

**Comment by lnicola** [comment]
Yeah, we support too many platforms to bundle a binary in the VSIX. There's some talk of that, but no progress AFAICT.

I do wonder if we can relax the version requirement, though: worst-case, if you're running a mismatched server, some of our LSP extensions will not work, but the basic functionality should still be there, just like if you were using another editor.
